Formula/dotnet.rb

```diff
@@ -67,74 +68,38 @@ class Dotnet < Formula
   # GCC builds have limited support via community.
   fails_with :gcc
 
-  # Apple Silicon build fails due to latest dotnet-install.sh downloading x64 dotnet-runtime.
-  # We work around the issue by using an older working copy of dotnet-install.sh script.
-  # Bug introduced with https://github.com/dotnet/install-scripts/pull/314
-  # TODO: Remove once script is fixed.
-  # Issue ref: https://github.com/dotnet/install-scripts/issues/318
-  resource "dotnet-install.sh" do
-    url "https://raw.githubusercontent.com/dotnet/install-scripts/dac53157fcb7e02638507144bf5f8f019c1d23a8/src/dotnet-install.sh"
-    sha256 "e96eabccea61bbbef3402e23f1889d385a6ae7ad84fe1d8f53f2507519ad86f7"
+  # Backport fix for error on aspnetcore version while building 'installer in tarball'.
+  # TODO: Remove when available in release.
+  # PR ref: https://github.com/dotnet/installer/pull/14938
+  patch do
+    url "https://raw.githubusercontent.com/Homebrew/formula-patches/f206f7a45b330cce79e6bfe9116fccd93b0d3ed8/dotnet/aspnetcore-version.patch"
+    sha256 "00103452e2f52831c04007f1b7f9fcd5ecddf0671943657104f0ac8d3a9ca613"
   end
 
-  # Fixes race condition in MSBuild.
-  # TODO: Remove with 6.0.3xx or later.
-  resource "homebrew-msbuild-patch" do
-    url "https://github.com/dotnet/msbuild/commit/64edb33a278d1334bd6efc35fecd23bd3af4ed48.patch?full_index=1"
-    sha256 "5870bcdd12164668472094a2f9f1b73a4124e72ac99bbbe43028370be3648ccd"
-  end
-
-  # Fix build failure on macOS due to missing ILAsm/ILDAsm
+  # Fix build failure on macOS due to missing bootstrap packages
   # Fix build failure on macOS ARM due to `osx-x64` override
   # Issue ref: https://github.com/dotnet/source-build/issues/2795
   patch :DATA
 
   def install
     ENV.append_path "LD_LIBRARY_PATH", Formula["icu4c"].opt_lib if OS.linux?
 
-    (buildpath/".dotnet").install resource("dotnet-install.sh")
-    (buildpath/"src/SourceBuild/tarball/patches/msbuild").install resource("homebrew-msbuild-patch")
-
-    Dir.mktmpdir do |sourcedir|
-      system "./build.sh", "/p:ArcadeBuildTarball=true", "/p:TarballDir=#{sourcedir}"
-      cd sourcedir
+    # The source directory needs to be outside the installer directory
+    (buildpath/"installer").install buildpath.children
+    cd "installer" do
+      system "./build.sh", "/p:ArcadeBuildTarball=true", "/p:TarballDir=#{buildpath}/sources"
+    end
 
+    cd "sources" do
```

Comment:
Just switched to this as a bit easier to debug when wanting to use `brew install --keep-tmp` unlike `Dir.mktmpdir` which gets deleted.